hazelcast/src/main/java/com/hazelcast/internal/nio/Packet.java

```diff
@@ -43,6 +43,8 @@
     // 1. URGENT (bit 4)
     // 2. Packet type (bits 0, 2, 5)
     // 3. Flags specific to a given packet type (bits 1, 6)
+    // 4. 4.x flag (bit 7)
```

Comment:
All packets in a stream will be of the same version. Why is this not part of the MemberHandshake?

Reply:
It is part of the `BindRequest` (`MemberHandshake`). If you're asking why we don't just set the version on the tcp connection and use that, I guess the problem is getting a reference to the connection in all places where the Packet is passed around, and it introduces coupling between the connection/network layer and other subsystems like operation handling.